### PR TITLE
datacube==1.8.1

### PR DESCRIPTION
**Pull request by Kirill888**
1. Bump datacube version to the latest, 1.8.1
2. Move installation of datacube libs into runner stage
3. Add mechanism to the Docker image to force re-installation of odc libs

The idea is to force docker cache invalidation by supplying:
```
--build-arg PURGE_CACHE_ODC=$(date)
```
when needed. This is not yet enabled in any github action, and hence untested.

Question is how to trigger this behaviour. Options are:

1. Branches with certain name patterns, for example if branch name starts with `update-`
2. Special hook that is triggered from outside with `curl`
3. PR comment?

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -43,9 +43,9 @@ RUN echo "Enable server extensions" \
   && jupyter serverextension enable --py --sys-prefix jupyterlab_iframe \
   && echo done
 
-COPY requirements-odc.txt /conf/
-RUN echo "Adding odc-dependencies" \
-  && env-build-tool extend /conf/requirements-odc.txt ${py_env_path}
+COPY requirements-odc-fixed.txt /conf/
```
Comment (alexgleith) [inline docker/Dockerfile]: Why `-fixed`?

Comment (alexgleith) [reply docker/Dockerfile]: Or does it mean 'fixed' as in 'not changing'? Maybe 'static' might be better... I read it as in fixed=repaired.

**Review by alexgleith — APPROVED** [review]
Seems reasonable.

File: docker/Dockerfile (hunk 1 above)
Comment (Kirill888) [reply docker/Dockerfile]: good point, will change to `-static`?

**Review by alexgleith — APPROVED** [review]
👍

**Pull request merged**